homeassistant/components/mqtt/binary_sensor.py

```diff
@@ -122,6 +122,7 @@ def __init__(
 
     async def mqtt_async_added_to_hass(self) -> None:
         """Restore state for entities with expire_after set."""
+        expire_after: int | None
```

Comment:
Maybe expire_after should be a class property, (re-)initialised on config update?
It seems that you are ready it from self._config multiple times.